component/espejote.jsonnet

```
@@ -0,0 +1,70 @@
+// main template for cm-hetznercloud
+local com = import 'lib/commodore.libjsonnet';
+local kap = import 'lib/kapitan.libjsonnet';
+local kube = import 'lib/kube.libjsonnet';
+local inv = kap.inventory();
+
+// The hiera parameters for the component
+local params = inv.parameters.espejote;
+
+// CRDs
+
+local crd = com.Kustomization(
+  'https://github.com/vshn/espejote/config/crd',
+  params.manifestVersion,
+);
+
+// Controller
+
+local patchEspejoteResources = {
+  patches+: [
+    {
+      patch: |||
+        - op: replace
+          path: "/spec/template/spec/containers/0/resources"
+          value: %s
+        - op: replace
+          path: "/spec/template/spec/containers/1/resources"
+          value: %s
+      ||| % [
+        std.manifestJsonMinified(params.resources.rbac_proxy),
+        std.manifestJsonMinified(params.resources.espejote),
+      ],
+      target: {
+        kind: 'Deployment',
+        name: 'controller-manager',
+        namespace: 'system',
+      },
+    },
+  ],
```

I have a very strong preference for `patchesStrategicMerge` as it will do the correct thing if containers are reordered or removed/added/renamed:

```suggestion
  patchesStrategicMerge+: [
    std.manifestJson({
      apiVersion: 'apps/v1',
      kind: 'Deployment',
      metadata: {
        name: 'controller-manager',
        namespace: 'system',
      },
      spec: {
        template: {
          spec: {
            containers: [
              {
                name: 'kube-rbac-proxy',
                resources: params.resources.rbac_proxy,
              },
              {
                name: 'manager',
                resources: params.resources.espejote,
              },
            ],
          },
        },
      },
    }),
  ],
```

---

Could do, allthough its deprecated it has never stopped me in the past..

---

It's un-deprecated AFAIK https://kubectl.docs.kubernetes.io/references/kustomize/builtins/#_patchesstrategicmerge_

Upstream uses it anywhere, and we use it even in this component for the namespace removal patch 😉